src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -759,21 +767,161 @@ public boolean isKnownDatasource( UUID studyId, String participantId, String dat
+                                                java.util.Optional.of( dstNeighborSetIds ),
+                                                java.util.Optional.empty()
+                                        )
+                                );
+
+                        if ( studyNeighbors.size() == 0 ) {
+                            logger.debug( "Attempt to remove participant without data." );
+                        }
+
+                        // create a key for all entity sets
+                        Sets.union( srcNeighborSetIds, dstNeighborSetIds ).forEach( entitySetId -> {
+                            toDeleteEntitySetIdEntityKeyId.put( entitySetId, new HashSet<>() );
+                        } );
+
+                        // fill Map<entitySetId, Set<entityKeyId>>
+                        studyNeighbors
+                                .get( participantEntityKeyId )
+                                .forEach( ( edgeEntitySetId, edgeNeighbor ) -> {
+                                    edgeNeighbor.forEach( ( neighborEntitySetId, neighborEntityIds ) -> {
+                                        Set<UUID> set = toDeleteEntitySetIdEntityKeyId.get( neighborEntitySetId );
+                                        set.add( neighborEntityIds.getNeighborEntityKeyId() );
+                                    } );
+                                } );
+                    }
+
+            );
+
+            // delete all neighbors
+            toDeleteEntitySetIdEntityKeyId
+                    .forEach(
+                            ( entitySetId, entityKeyId ) -> chronicleDataApi
+                                    .deleteEntities( entitySetId, entityKeyId, deleteType )
+                    );
+
+            // delete participants
+            Integer deleted = userDataApi.deleteEntities( participantsEntitySetId, participantsToRemove, deleteType );
+            logger.info("Deleted "+ deleted.toString() +" entities for participant "+participantId.toString() + ".");
+
+            // delete study if no participantId is specified
+            if ( participantId.isEmpty() ) {
+                // delete participant entity set
+                userEntitySetsApi.deleteEntitySet( participantsEntitySetId );
+                logger.info("Deleted participant dataset for study " + studyId +".");
+                List<Map<FullQualifiedName, Set<Object>>> studySearchResult = userSearchApi
+                        .executeEntitySetDataQuery( studyESID,
+                                new SearchTerm( studyId.toString(), 0, SearchApi.MAX_SEARCH_RESULTS,
+                                        java.util.Optional.of( false ) ) )
+                        .getHits();
+                if ( studySearchResult.size() != 1 ) {
+                    logger.error( "Tried to remove study " + studyId
+                            + "but couldn't find matching study in chronicle_study." );
+                }
+
```

Comment:
I don't think you need this search api call. You could simply do`getStudyEntityKeyId(studyId)`.